pallets/parachain-staking/src/lib.rs

```diff
@@ -1707,80 +1466,28 @@ pub mod pallet {
 		/// their chances to be included in the set of candidates in the next
 		/// rounds.
 		///
-		/// Emits `DelegatorLeft`.
+		/// Automatically increments the accumulated rewards of the origin of
+		/// the current delegation.
 		///
-		/// # <weight>
-		/// Weight: O(C) where C is the number of delegations for this delegator
-		/// which is bounded by by `MaxCollatorsPerDelegator`.
-		/// - Reads: [Origin Account], DelegatorState, BlockNumber, Unstaking,
-		///   TopCandidates, MaxSelectedCandidates, C * CandidatePool,
-		/// - Writes: Unstaking, CandidatePool, TotalCollatorStake,
-		/// - Kills: DelegatorState
-		/// # </weight>
+		/// Emits `DelegatorLeft`.
 		#[pallet::weight(<T as pallet::Config>::WeightInfo::leave_delegators(
 			T::MaxTopCandidates::get(),
 			T::MaxDelegatorsPerCollator::get()
 		))]
 		pub fn leave_delegators(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
 			let acc = ensure_signed(origin)?;
 			let delegator = DelegatorState::<T>::get(&acc).ok_or(Error::<T>::DelegatorNotFound)?;
-			let num_delegations: u32 = delegator.delegations.len().saturated_into();
-			for stake in delegator.delegations.into_iter() {
-				Self::delegator_leaves_collator(acc.clone(), stake.owner.clone())?;
-			}
+			// should never throw
+			let collator = delegator.owner.ok_or(Error::<T>::DelegationNotFound)?;
+			Self::delegator_leaves_collator(acc.clone(), collator)?;
```

Comment:
I moved the update of `InflationInfo` for the two extrinsics (`set_inflation` and `adjust_yearly_inflation_rate`) to a helper function and added [more documentation explaining why this can be called by any signed origin](https://github.com/KILTprotocol/kilt-node/pull/384/commits/f459050e76ebed0c6d142a70d13a9f66730d60e5#diff-842251b834245800808bd740184c1e3b18c6af1f95fb1bd4d484420017fdf86eR1727-R1729).

Reply:
[55c6ac3](https://github.com/KILTprotocol/kilt-node/pull/384/commits/55c6ac30dd1771e0cd36ee6fa80e01475d912870)